### ISSUE TITLE
ConfigMap change does not trigger bean refresh

### ISSUE THREAD

**Issue by martin-schaefer** [spring-cloud/spring-cloud-kubernetes#402]
I am doing some training with spring-cloud-kubernetes on Red Hat minishift. I managed to define a ConfigMap which is correctly found and the configured message ${service.message} is shown..

However, when I change the value in the ConfigMap, the RestController in the @RefreshScope seems not to be refreshed and the message stays the same. Only a restart of the app updates the message.

According to the documentation I assumed this should work. Did I miss some additional step or is this not working yet?

```java
@Controller
@RefreshScope
public class RestController {

	private final String message;

	public RestController(@Value("${service.message}") String message) {
		this.message = message;
	}

	/**
	 * Say hello to earth
	 * 
	 * @return First word from the moon
	 * 
	 */
	@GetMapping("/hello")
	@ResponseBody
	public String hello() {
		return message + " (" + LocalDateTime.now().toString() + ")";
	}
}
```
The ConfigMap

```yaml
apiVersion: v1
data:
  application.yml: |-
    service:
      message: Say hello to the world  
kind: ConfigMap
metadata:
  creationTimestamp: '2019-06-02T17:35:24Z'
  name: openshift-spring-boot-service
  namespace: dsp-dev
  resourceVersion: '67065'
  selfLink: /api/v1/namespaces/dsp-dev/configmaps/openshift-spring-boot-service
  uid: cd7a8217-855c-11e9-8dc5-00155d010b07
```

**Comment by geoand** [comment]
I would suggest you take a look at [this](https://github.com/spring-cloud/spring-cloud-kubernetes/tree/master/spring-cloud-kubernetes-integration-tests/simple-configmap) integration test. My guess is that you probably haven't enabled restart.

**Comment by martin-schaefer** [comment]
Thanks for the hint. I activated the config reload like this:

```yaml
spring:
  cloud:
    kubernetes:      
      reload:
        enabled: true
```
But then I received the following error on startup:
```
Description:
Parameter 2 of method configurationUpdateStrategy in org.springframework.cloud.kubernetes.config.reload.ConfigReloadAutoConfiguration$ConfigReloadAutoConfigurationBeans required a bean of type 'org.springframework.cloud.context.restart.RestartEndpoint' that could not be found.
The following candidates were found but could not be injected:
	- Bean method 'restartEndpoint' in 'RestartEndpointWithIntegrationConfiguration' not loaded because @ConditionalOnClass did not find required class 'org.springframework.integration.monitor.IntegrationMBeanExporter'
	- Bean method 'restartEndpointWithoutIntegration' in 'RestartEndpointWithoutIntegrationConfiguration' not loaded because @ConditionalOnEnabledEndpoint no property management.endpoint.restart.enabled found so using endpoint default
Action:
Consider revisiting the entries above or defining a bean of type 'org.springframework.cloud.context.restart.RestartEndpoint' in your configuration.
```

Then I decided to activate refresh/restart and to expose all actuators as shown below. This way it works. However, it feels a bit strange to expose the restart endpoint even tough I left the value `spring.cloud.kubernetes.reload.mode` to its default (`event`)

```yaml
management:
  endpoint:
    refresh:      
      enabled: true
    restart: 
      enabled: true
  endpoints:
    web:
      exposure:
        include: "*"
```

**Comment by geoand** [comment]
@spencergibb @ryanjbaxter is this consistent with the rest of Spring Cloud?

**Comment by ryanjbaxter** [comment]
No it is not consistent.  Why would the restart endpoint need to be enabled when we are using refresh by default?

**Comment by Haybu** [comment]
this is done part of #440, merged into 1.0.x. I would close this one.
